File: 탐욕법/큰_수_만들기/solve.cpp
#include <string>
#include <vector>
#include <iostream>
#include <algorithm>
#include <queue>
#include <set>

using namespace std;

string solution(string number, int k) { 
    string answer = "";

    int idx = -1;
    for(int i=0; i<number.size()-k; i++){
        char ch = '0';
        
        for(int j=idx+1; j<=i+k; j++){
    
            if(number[j] > ch){
                idx = j;
                ch = number[j];
            }
        }
        
        answer += ch;
    }
    
    return answer;
}

File: 탐욕법/섬_연결하기/solve.py
from queue import PriorityQueue

edge = []
sets = []
pq = PriorityQueue()

def findParent(x):
    global sets
    if sets[x] == x:
        return x
    else :
        sets[x] = findParent(sets[x])
        return sets[x]

def unionTwo(x, y):
    global sets
    
    x = findParent(x)
    y = findParent(y)
    sets[y] = x
    
    
def solution(n, costs):
    global edge, sets, pq
    
    answer = 0
    
    sets = [i for i in range(n)]

    for cost in costs:
        pq.put([cost[2], cost[0], cost[1]])
        
    while not pq.empty():
        cur = pq.get()
        w = cur[0]
        left = cur[1]
        right = cur[2]
        
        if findParent(left) != findParent(right):
            unionTwo(left, right)
            answer += w
        
    return answer
